WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift

```diff
@@ -62,7 +62,7 @@ extension NewOrderHostingController {
 ///
 extension NewOrderHostingController: UIAdaptivePresentationControllerDelegate {
     func presentationControllerShouldDismiss(_ presentationController: UIPresentationController) -> Bool {
-        return !viewModel.hasChanges
+        return viewModel.canBeDismissed || !viewModel.hasChanges
```

Comment:
I think this has a bit of problems as an edited order with changes will trigger an alert and a created order with changes won't trigger the alert. 

https://user-images.githubusercontent.com/562080/175144418-ae06dc20-2313-4d90-9604-fa74d3dc575c.mov

Reply:
I fixed this by updating how `canBeDismissed` work https://github.com/woocommerce/woocommerce-ios/pull/7065/commits/a3f7913622b7455a000b2a058c0a1c1761654add  

Basicaly, it follows these rules:
-  Creation can be dismissed when there aren't changes pending to commit.
-  Editing can always be dismissed because changes are committed instantly.

Does that make sense?

Reply:
Looks like it should be `flow != .creation` initially. But your refactor reads better and hides details in viewmodel 👍